packages/oats/src/driver.ts

```diff
@@ -42,6 +42,7 @@ export interface Driver {
     security?: UnsupportedFeatureBehaviour;
   };
   forceGenerateTypes?: boolean; // output the type file even if it would have been already generated
+  enableTypeManipulation?: boolean; // manipulate generation of structural and nominal types
```

Comment:
could we have this as `typename?: (kind: shape | value, name: string) => string` as we are not actually doing anything else than renaming the types. We could also maybe wrap oautil.typenamify to do the type mapping as needed to make the code a bit less invasive.

another question is how this should work when importing types from other files? Having this setting in one file requires similar setting when generating the other files.

Reply:
one way to do this would be to hook this up with the name resolvers if needed. If there is need for more finessing we could I guess require writing their own resolver to deal with the name mapping. However in any case I don't think we should not after resolution do any mapping with the name